validphys2/src/validphys/commondataparser.py

```diff
@@ -40,22 +108,825 @@
+        """ObservableMetaData is defined by:
+        - the setname
+        - the variant used
+        - the data
+        """
+        return hash((self.name, self.applied_variant, self.data_central))
+
+    def check(self):
+        """Various checks to apply manually to the observable before it is used anywhere
+        These are not part of the __post_init__ call since they can only happen after the metadata
+        has been read and the observable selected.
+        """
+        # Check that the data_central is empty if and only if the dataset is a positivity/integrability set
+        if self.data_central is None and not self.is_lagrange_multiplier:
+            raise ValidationError(f"Missing `data_central` field for {self.name}")
+
+        # Check that plotting.plot_x is being filled
+        if self.plotting.plot_x is None:
+            ermsg = f"No variable selected as x-axis in the plot for {self.name}. Please add `plotting::plot_x`."
+            if self.plotting.x is not None:
+                ermsg += "Please replace `plotting::x` with `plotting::plot_x`."
+            raise ValidationError(ermsg)
+
+        # Ensure that all variables in the kinematic coverage exist
+        for var in self.kinematic_coverage:
+            if var not in self.kinematics.variables:
+                raise ValidationError(
+                    f"Variable {var} is in `kinematic_coverage` but not included in `kinematics` for {self.name}"
+                )
+
+        if len(self.kinematic_coverage) > 3:
+            raise ValidationError(
+                "Only a maximum of 3 variables can be used for `kinematic_coverage`"
+            )
+
+    def apply_variant(self, variant_name):
+        """Return a new instance of this class with the variant applied
+
+        This class also defines how the variant is applied to the commondata
+        """
+        try:
+            variant = self.variants[variant_name]
+        except KeyError as e:
+            raise ValueError(f"The requested variant does not exist {self.variant_name}") from e
+
+        variant_replacement = {}
+        if variant.data_uncertainties is not None:
+            variant_replacement["data_uncertainties"] = variant.data_uncertainties
+        if variant.theory is not None:
+            variant_replacement["theory"] = variant.theory
+        if variant.data_central is not None:
+            variant_replacement["data_central"] = variant.data_central
+
+        return dataclasses.replace(self, applied_variant=variant_name, **variant_replacement)
+
+    @property
+    def is_positivity(self):
+        return self.setname.startswith("NNPDF_POS")
+
+    @property
+    def is_integrability(self):
+        return self.setname.startswith("NNPDF_INTEG")
+
+    @property
+    def is_lagrange_multiplier(self):
+        return self.is_positivity or self.is_integrability
+
+    @property
+    def path_data_central(self):
+        return self._parent.folder / self.data_central
```

Comment:
```suggestion
```
Do we really need this to be a separate attribute (same for `paths_uncertainties`)? This is only used in `load_data_central` function and therefore can be defined there.

Reply:
While in vp it is not used, it is the only way to have access to the actual path which might be useful (for instance, if from pineko you want to read the kinematics or data without loading them) since the metadata itself only contains the name of the file but not the full path (it is also the reason we cannot check for existence while parsing).

Reply:
Ah, then it's ok to leave these as they are.